File: langwatch/src/features/onboarding/regions/observability/codegen/snippets/typescript/openai.snippet.ts
import { setupObservability } from "langwatch/observability/node";
import { getLangWatchTracer } from "langwatch";
import { OpenAI } from "openai";

setupObservability({ serviceName: "<project_name>" }); // +

const tracer = getLangWatchTracer("<project_name>"); // +

async function main(message: string): Promise<string> {
  const openai = new OpenAI();

  return await tracer.withActiveSpan("main", async span => { // +
    span.setInput(message); // +

    const response = await openai.chat.completions.create({
      model: "gpt-5",
      messages: [{ role: "user", content: message }],
    });

    const text = response.choices[0].message.content as string;
    span.setOutput(text); // +
    return text;
  }); // +
}

console.log(await main("Hey, tell me a joke"));

File: platform/app/src/features/onboarding/regions/observability/codegen/snippets/python/openai.snippet.py
import langwatch  # +
from openai import OpenAI

langwatch.setup()  # +
client = OpenAI()


@langwatch.trace(name="OpenAI Chat Completion")
def get_openai_chat_response(user_prompt: str):
    langwatch.get_current_trace().autotrack_openai_calls(client)  # +

    response = client.chat.completions.create(
        model="gpt-5",
        messages=[{"role": "user", "content": user_prompt}],
    )
    completion = response.choices[0].message.content
    return completion


if __name__ == "__main__":
    user_query = "Tell me a joke"
    response = get_openai_chat_response(user_query)

    print(f"User: {user_query}")
    print(f"AI: {response}")
